plugin/src/main/kotlin/org/gradle/github/dependency/extractor/internal/DependencyExtractorService.kt

```diff
@@ -78,55 +76,63 @@ abstract class DependencyExtractorService :
         tailrec fun recursivelyExtractProjects(projects: Set<LoadProjectsBOT.Result.Project>) {
             if (projects.isEmpty()) return
             projects.forEach { project ->
-                gitHubRepositorySnapshotBuilder.addProject(details.buildPath, project.path, project.buildFile)
+                gitHubRepositorySnapshotBuilder.addProject(project.identityPath, project.buildFile)
             }
             val newProjects = projects.flatMap { it.children }.toSet()
             recursivelyExtractProjects(newProjects)
         }
         recursivelyExtractProjects(setOf(result.rootProject))
     }
 
-    private fun extractDependencies(
-        details: ResolveConfigurationDependenciesBOT.Details,
-        result: ResolveConfigurationDependenciesBOT.Result
+    private fun extractConfigurationDependencies(
+        details: ResolveConfigurationDependenciesBuildOperationType.Details,
+        result: ResolveConfigurationDependenciesBuildOperationType.Result
     ) {
-        val rootComponentId = result.rootComponent.id
-        val trueProjectPath =
-            (rootComponentId as? ProjectComponentIdentifier)?.projectPath
-                ?: details.projectPath
+        val repositoryLookup = RepositoryUrlLookup(details, result)
```

Comment:
I intentionally built this object after the `if` because then the memory wouldn't be allocated for it if we were just going to immediately `return`.

Reply:
Good point. I'll defer this instantiation until it's required.